server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1834,6 +1836,42 @@ public void syncDatastoreClusterStoragePool(long datastoreClusterPoolId, List<Mo
         handleRemoveChildStoragePoolFromDatastoreCluster(childDatastoreUUIDs);
     }
 
+    /**
+     * fixed mismatching between db uuids and and custom
+     * attribute uuids
+     *
+     * To different formats of uuids exists
+     * 1. xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+     * 2. xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+     * @param uuid of existing pool
+     * @return existing pull or null otherwise
+     */
+    private StoragePoolVO getExistingPoolByUuid(String uuid){
+        StoragePoolVO storagePool = _storagePoolDao.findByUuid(uuid);
+        if(storagePool != null){
+            return storagePool;
+        }
+
+        //this case is unlikely (DB uuid without separators), but safety first
```

Comment:
@DK101010 I feel it is not correct to find the pool with the mismatched UUID (by removing the separators).

the UUIDs format in DB should be always with separators (`xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`). It is better to fix if there is any mismatch / discrepancy in populating the UUIDs anywhere.

Reply:
@sureshanaparti ok, I already had the assumption that it should be like that. 
We can set the correct uuid in the method here. However, this would mean that all primary storages that were created with a datastore cluster (4.15.1) are created again (with the correct uuid) when they are included again.

Reply:
@sureshanaparti I just saw that the method is also called during the start of cloudstack. So as soon as cloustack starts, the primary storages in question are recreated.

Reply:
> @sureshanaparti ok, I already had the assumption that it should be like that.
> We can set the correct uuid in the method here. However, this would mean that all primary storages that were created with a datastore cluster (4.15.1) are created again (with the correct uuid) when they are included again.

@DK101010 If the UUID for any storage pool in datastore cluster is not inline with the actual UUID format used across, that UUID of these storage pool  have to be fixed, instead looking for storage pool with other format.

Reply:
I agree with @sureshanaparti, a non conforming UUID is actually a corruption of data and should be fixed instead of worked around, @DK101010 .

Reply:
@sureshanaparti @DaanHoogland 
I think I have expressed myself a little unclearly. Currently all custom attributes of the storage pools have a uuid without a separator (at least in the Vmware context). I think this is intentional even if I don't know why. The syncDatastoreClusterStoragePool method writes these uuids into the database without customising them.

I understand and am with you. 

I can modify the method to only store uuids with separators in the database. But this means that we still need to upgrade the database to match existing uuids of primary storages, otherwise new primary storages will be created. 

However, I don't know of any sql that can convert a uuid without separators into a uuid with separators.

Reply:
@DK101010 the upgrade includes a java step. it can be implemented in that.

Reply:
@DaanHoogland That's new for me. Where I can do that?

Reply:
look for `Upgrade41510to41600.java`